### PR TITLE
file_utils: get_tool_path to always return an absolute path

### PR DESCRIPTION
**Pull request by sergiusens**
Change get_tool_path to always return an absolute path and fail if
the required command cannot be found.

The unit tests have been adapted to not be path dependent and tests
for retrieving environment dependent paths have been added.

LP: #1785320

Signed-off-by: Sergio Schvezov <sergio.schvezov@canonical.com>

- [x] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [x] If this is a bugfix. Have you checked that there is a bug report open for the issue you are trying to fix on [bug reports](https://bugs.launchpad.net/snapcraft)?
- [x] If this is a new feature. Have you discussed the design on the [forum](https://forum.snapcraft.io)?
- [x] Have you successfully run `./runtests.sh static`?
- [x] Have you successfully run `./runtests.sh unit`?

-----

### CODE REVIEW

File: snapcraft/internal/errors.py [hunk 1]
```diff
@@ -295,6 +295,17 @@ def __init__(self, message):
         super().__init__(message=message)
 
 
+class ToolMissingError(SnapcraftError):
+
+    fmt = (
+        "A tool snapcraft depends on could not be found: {command_name!r}.\n"
+        "Ensure the tool is installed and available, and try again."
```
Comment (kyrofa) [inline snapcraft/internal/errors.py]: This is a workaround, but in reality, for all the use-cases of this function so far (e.g. xdelta3, mksquashfs, etc.) all of these tools should ALWAYS be found, and if they're not, it's a bug we should know about. Worth adding "please log a bug" to this?

File: tests/unit/commands/test_push.py [hunk 2]
```diff
@@ -79,64 +77,6 @@ def test_push_a_snap(self):
         )
         mock_upload.assert_called_once_with("basic", self.snap_file)
 
-    def test_push_a_snap_running_from_snap(self):
```
Comment (kyrofa) [inline tests/unit/commands/test_push.py]: This test seems useful, is it duplicated somewhere?

**Comment by sergiusens** [comment]
That last change enables this -> https://asciinema.org/a/hGSsKKhmPSPA92l8LyVDBAiBg

File: tests/unit/commands/test_push.py (hunk 2 above)
Comment (kyrofa) [reply tests/unit/commands/test_push.py]: We clarified this offline: the tests covering get_tool_path here cover most of what is removed here. We need a suite that runs in docker to cover everything, which we can introduce later.

**Comment by sergiusens** [comment]
spread 17.10 tests failed, ignoring those.

**Pull request merged**